From: James Smart <jsmart2021@gmail.com>
Date: Tue, 23 Oct 2018 13:41:02 -0700
Subject: [PATCH] scsi: lpfc: Fix lpfc_sli4_read_config return value check
Git-commit: 3952e91f110b1abd1f139a04896c3ba66171df84
Patch-mainline: v5.0-rc1
References: bsc#1114015

An error is an error - but not to the existing return value check.

Revise check to handle any failure, not just EIO.

Signed-off-by: Dick Kennedy <dick.kennedy@broadcom.com>
Signed-off-by: James Smart <jsmart2021@gmail.com>
Reviewed-by: Hannes Reinecke <hare@suse.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/lpfc/lpfc_init.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/lpfc/lpfc_init.c b/drivers/scsi/lpfc/lpfc_init.c
index df0ec52a9b56..7f63d32c3589 100644
--- a/drivers/scsi/lpfc/lpfc_init.c
+++ b/drivers/scsi/lpfc/lpfc_init.c
@@ -4896,11 +4896,11 @@ lpfc_sli4_async_sli_evt(struct lpfc_hba *phba, struct lpfc_acqe_sli *acqe_sli)
 
 		/* Issue READ_CONFIG mbox command to refresh supported speeds */
 		rc = lpfc_sli4_read_config(phba);
-		if (rc == -EIO) {
+		if (rc) {
 			phba->lmt = 0;
 			lpfc_printf_log(phba, KERN_ERR, LOG_SLI,
 					"3194 Unable to retrieve supported "
-					"speeds\n");
+					"speeds, rc = 0x%x\n", rc);
 		}
 		vports = lpfc_create_vport_work_array(phba);
 		if (vports != NULL) {
-- 
2.16.4

